pipeline/relation_embedder/router/src/main/scala/weco/pipeline/router/RouterWorkerService.scala

```diff
@@ -33,16 +33,24 @@ class RouterWorkerService[MsgDestination](
             pipelineStream.config,
             work => Future.fromTry(processMessage(work))))
     )
+  }
 
-  private def processMessage(w: Work[Merged]): Try[List[Work[Denormalised]]] =
-    w.data.collectionPath match {
-      case None =>
-        val denormalisedWork = w.transition[Denormalised](
-          (Relations.none, Set.empty)
-        )
-        Success(List(denormalisedWork))
-
-      case Some(CollectionPath(path, _)) =>
-        pathsMsgSender.send(path).map(_ => List())
+  private def processMessage(
+    work: Work[Merged]): Try[List[Work[Denormalised]]] = {
+    (work.data.collectionPath, work.state.relations) match {
+      // For TEI works relations are already populated based on
+      // inner works extracted by the TEI transformer. We don't need
+      // to repopulate them in the relation embedder.
+      // We don't expect TEI works to have a collectionPath field populated.
+      case (None, relations) =>
+        Success(List(work.transition[Denormalised]((relations, Set.empty))))
+      case (Some(CollectionPath(path, _)), relations)
+          if relations == Relations.none =>
```

Comment:
Does this work? I'm guessing there's a reason you didn't do it this way, but just checking.

```suggestion
      case (Some(CollectionPath(path, _)), Relations.none) =>
```

Reply:
This doesn't compile. `Relations.none` is not a type so you can't pattern match. It's not particularly nice, I agree. I tried making `Relations` an `Option` but I suspected it wouldn't work otherwise Nick would have done it. It ended up being very convoluted and I backtracked